torch/nativert/executor/DelegateExecutor.cpp

```diff
@@ -0,0 +1,65 @@
+#include <torch/nativert/executor/DelegateExecutor.h>
+
+#ifndef _WIN32
+#include <unistd.h>
+#endif
+
+#include <c10/util/Logging.h>
+
+#include <c10/util/string_view.h>
+#include <torch/nativert/common/FileUtil.h>
+
+namespace torch::nativert {
+
+namespace {
+char* _mkdtemp(char* outputDir) {
+  // mkdtemp is not available on Windows
+#ifdef _WIN32
+  return nullptr;
+#else
+  return mkdtemp(outputDir);
+#endif
+}
+} // namespace
+
+std::string extractToTemporaryFolder(
+    caffe2::serialize::PyTorchStreamReader& packageReader,
+    const std::string& targetPath) {
+  char outputDir[] = "/tmp/model_XXXXXX";
```

Comment:
@yiming0416 seems the directory name here is a bit rough, could we do something better here? (e.g. use a more descriptive name + a random uuid)